File: euler_method.py
def euler_method(nodes_num:int, start:float, end:float, init_value:float, fn:callable):
    # 默认所有输入数据都是合法的
    
    # 计算步长
    h = (end - start) / nodes_num
    # 定义点列
    points = [start + i*h for i in range(nodes_num)]
    values = [0 for _ in range(nodes_num)]
    
    # 开始迭代计算values
    index = 1
    values[0] = init_value
    while index < nodes_num:
        values[index] = values[index - 1] + h * fn(points[index - 1], values[index - 1])
        index += 1
    return values

fn = lambda x,y: x*y

results = euler_method(100, 0, 1, 1, lambda x,y: x+y)
print(results)

File: 《微分方程的数值解法与程序实现》-程序算例/euler_method.cpp
#include <iostream>
#include <math.h>

static float f(float x, float y)
{
	return y - (2 * x / y);
}

void main1()
{
	// 给区间分段，插入的分点数量为n
	const int n = 10;
	// 区间起点和终点
	float start, end;
	start = 0;
	end = 1;
	// 步长
	float h = (end - start) / (n - 1);

	// 生成点列
	float points[n] = { 0 };
	float values[n] = { 0 };
	int index = 1;
	float init_value = 1;
	for (int i = 0; i < n; i++)
	{
		// 这一步是生成点列
		points[i] = start + i * h;
	}
	// 设置初值
	values[0] = init_value;
	// 迭代计算其他值
	while (index < n)
	{
		values[index] = values[index - 1] + h * f(points[index - 1], values[index - 1]);
		index += 1;
	}

	for (int i = 0; i < n; i++)
	{
		std::cout << "(" << points[i] << "," << values[i] << ")" << "\n";
	}
}